app/components/PasswordResetForm.tsx

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+import { redirect } from '@remix-run/node'
+import { Form } from '@remix-run/react'
+import { Button, Label, TextInput } from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+import { PasswordManager } from '../routes/user/password.server'
+import { getFormDataString } from '~/lib/utils'
+
+export async function handlePasswordActions(
+  request: Request,
+  redirectSource: string
+) {
+  const data = await request.formData()
+  const oldPassword = getFormDataString(data, 'oldPassword')
+  const newPassword = getFormDataString(data, 'newPassword')
+  const confirmPassword = getFormDataString(data, 'confirmPassword')
+
+  if (!oldPassword || !newPassword || !confirmPassword) {
+    throw new Response('all password fields must be present', { status: 400 })
+  }
+
+  if (newPassword != confirmPassword) {
+    throw new Response('passwords must match', { status: 400 })
+  } else {
+    const passwordManager = await PasswordManager.create(
+      request,
+      oldPassword,
+      newPassword
+    )
+    await passwordManager.updatePassword()
+    return redirect('/user')
```

Comment:
Rather than redirecting the user, what if we display  "Saved" checkmark next to the submit button like we do on some other forms?

Reply:
got this.